### DISCUSSION TITLE
GraphQL Query to fully expand a hierarchical property

### DISCUSSION THREAD

**Discussion by bo55vxr** [ChilliCream/hotchocolate#5110] [Q&A] [answered]
I am putting a Graph QL layer over an API which has within its response a _recursive_ list as a hierarchy for location.

Definition of Location class
```
public class Location 
{
 public string LocationID {get; set;}
 public string LocationName {get; set;}
 public IEnumerable<Location> Children {get; set;}
}
```

As you can see, this can effectively go infinitely deep, but in reality it wont, but I have no idea how deep it would go.

How do I write a GraphQL request to fully expand each location

**Comment by PascalSenn** [comment, marked as answer]
@bo55vxr  this is not possible in GraphQL. Recursive structures have to be fetched in batches.
You can e.g. fetch the first three layers and then execute another query to fetch another three layers based on the Ids
